internal/constants/scripts.go

```diff
@@ -196,3 +195,27 @@ if [ "$host_dir" ]
 fi
 echo "PreNetworkConfig End"
 `
+
+const PreNetworkConfigService = `
+[Unit]
+Description=Assisted static network config
+DefaultDependencies=no
+
+# We need to run after the kernel and udev have detected all the network interface devices,
+# otherwise the mechanism that we use to find the MAC addresses will not work and the configuration
+# will not be applied.
+After=systemd-udev-settle.service
+
+# We need to run after this service because it runs the 'nm-initrd-generator' tool, which removes
+# all the connections and re-creates them from kernel command line arguments.
+After=coreos-enable-network.service
```

Comment:
Hmm...I think this one is kind of an implementation detail that may change in the future, but probably not the near future.

Reply:
Yes, I saw that explained in the service file, but didn't find a way around this. If I don't do it the that service re-creates the "default_connection", NetworkManager tries to activate it and then it introduces the 90 seconds wait till DHCP fails.